mne/channels/interpolation.py

```diff
@@ -213,3 +213,62 @@ def _interpolate_bads_meg(inst, mode='accurate', origin=(0., 0., 0.04),
     info_to = pick_info(inst.info, picks_bad)
     mapping = _map_meg_channels(info_from, info_to, mode=mode, origin=origin)
     _do_interp_dots(inst, mapping, picks_good, picks_bad)
+
+
+@verbose
+def _interpolate_bads_nirs(inst, method='nearest', verbose=None):
+    """Interpolate bad nirs channels. Simply replaces by closest non bad.
+
+    Parameters
+    ----------
+    inst : mne.io.Raw, mne.Epochs or mne.Evoked
+        The data to interpolate. Must be preloaded.
+    method : str
+        Only the method 'nearest' is currently available. This method replaces
+        each bad channel with the nearest non bad channel.
+    %(verbose)s
+    """
+    from scipy.spatial.distance import pdist, squareform
+    from mne.preprocessing.nirs import _channel_frequencies,\
+        _check_channels_ordered
+
+    freqs = np.unique(_channel_frequencies(inst))
+    # Returns pick of all nirs and ensures channels are correctly ordered
+    picks_nirs = _check_channels_ordered(inst, freqs)
+    nirs_ch_names = [inst.info['ch_names'][p] for p in picks_nirs]
+    bads_nirs = [ch for ch in inst.info['bads'] if ch in nirs_ch_names]
+
+    # select the bad meg channel to be interpolated
+    if len(bads_nirs) == 0:
+        picks_bad = []
+    else:
+        picks_bad = pick_channels(inst.info['ch_names'], bads_nirs,
+                                  exclude=[])
+    bads_mask = [p in picks_bad for p in picks_nirs]
+
+    chs = [inst.info['chs'][i] for i in picks_nirs]
+    locs3d = np.array([ch['loc'][:3] for ch in chs])
+    dist = pdist(locs3d)
+    dist = squareform(dist)
+
+    if method == 'nearest':
+
+        for bad in picks_bad:
+            dists_to_bad = dist[bad]
+            # Ignore distances to self
+            dists_to_bad[dists_to_bad == 0] = np.inf
+            # Ignore distances to other bad channels
+            dists_to_bad[bads_mask] = np.inf
+            # Find closest remaining channels
+            closest_idx = np.where(dists_to_bad == np.amin(dists_to_bad))[0]
+            # Return the same frequency
+            closest_idx = closest_idx[bad % 2]
+            inst._data[bad] = inst._data[closest_idx]
+
+        inst.info['bads'] = []
+
+    else:
+        warn('No interpolation applied. Unknown NIRS interpolation '
+             'method: ' + method)
```

Comment:
This should be an error instead of a warning. You can just replace it with (before the conditional):
```
_check_option('method', method, ['nearest'])
```